src/platform/ESP32/bluedroid/BLEManagerImpl.cpp

```diff
@@ -907,6 +930,25 @@ CHIP_ERROR BLEManagerImpl::MapBLEError(int bleErr)
     }
 }
 
+void BLEManagerImpl::CancelBleAdvTimeoutTimer(void)
+{
+    if (SystemLayer().IsTimerActive(BleAdvTimeoutHandler, nullptr))
```

Comment:
Why do you need the IsTimerActive bit?   CancelTimer can be called even if the timer is not running and will just do nothing...